src/spesh/deopt.c

```diff
@@ -162,19 +162,19 @@ static void uninline(MVMThreadContext *tc, MVMFrame *f, MVMSpeshCandidate *cand,
 }
 
 static void deopt_named_args_used(MVMThreadContext *tc, MVMFrame *f) {
-    if (f->spesh_cand->deopt_named_used_bit_field)
-        f->params.named_used.bit_field = f->spesh_cand->deopt_named_used_bit_field;
+    if (f->spesh_cand->body.deopt_named_used_bit_field)
+        f->params.named_used.bit_field = f->spesh_cand->body.deopt_named_used_bit_field;
 }
 
 /* Materialize an individual replaced object. */
 static void materialize_object(MVMThreadContext *tc, MVMFrame *f, MVMObject ***materialized,
                                MVMuint16 info_idx, MVMuint16 target_reg) {
     MVMSpeshCandidate *cand = f->spesh_cand;
     if (!*materialized)
-        *materialized = MVM_calloc(MVM_VECTOR_ELEMS(cand->deopt_pea.materialize_info), sizeof(MVMObject *));
+        *materialized = MVM_calloc(MVM_VECTOR_ELEMS(cand->body.deopt_pea.materialize_info), sizeof(MVMObject *));
     if (!(*materialized)[info_idx]) {
-        MVMSpeshPEAMaterializeInfo *mi = &(cand->deopt_pea.materialize_info[info_idx]);
-        MVMSTable *st = (MVMSTable *)cand->spesh_slots[mi->stable_sslot];
+        MVMSpeshPEAMaterializeInfo *mi = &(cand->body.deopt_pea.materialize_info[info_idx]);
+        MVMSTable *st = (MVMSTable *)cand->body.spesh_slots[mi->stable_sslot];
         MVMP6opaqueREPRData *repr_data = (MVMP6opaqueREPRData *)st->REPR_data;
         MVMROOT(tc, f, {
             MVMObject *obj = MVM_gc_allocate_object(tc, st);
```

Comment:
`cand` can become outdated here (and probably we need to check in the callers about this also)

Reply:
You (just) mean it needs to be added to the MVMROOT?

Reply:
Yes, here, but also check if there's a caller of this function that holds a pointer to `cand`, and that would thus require an `MVMROOT` in the caller also.